packages/peregrine/lib/talons/Gallery/useGalleryItem.js

```diff
@@ -18,17 +18,48 @@ export const useGalleryItem = props => {
     const { url_key } = item;
 
     // Prefetch the underlying product for this gallery item, but only if we
-    // don't have any product data yet.
+    // don't have any product data yet, and only once the gallery item is in
+    // view.
     const [runPrefetchQuery] = useLazyQuery(prefetchProductQuery);
-    useEffect(() => {
-        if (item) {
-            runPrefetchQuery({
-                variables: {
-                    urlKey: url_key,
-                    onServer: false
-                },
-                fetchPolicy: 'cache-first'
+    const galleryItemRef = useRef();
+
+    const handleScrollIntoView = useCallback(
+        entries => {
+            entries.forEach(entry => {
+                if (entry.isIntersecting) {
```

Comment:
Test this by adding items beyond the visible viewport. I did it by using a mobile view in dev tools and increased the gap between the gallery rows.

Reply:
Makes sense to me. Maybe this clause should avoid executing more than once for the same element?

```es6
if (!hasPrefetched && entry.isIntersecting)
```